### PR TITLE
feat(spanv2): Add http attribute normalization

### PR DESCRIPTION
**Pull request by Ahmed-Labs**
Ports over http attribute normalization logic from the transactions span pipeline. The actual logic for normalization is extracted from the SpanV1 [tag extraction function](https://github.com/getsentry/relay/blob/7cb7f0096bdc50b7e1b7ad22f8100a779b1279a3/relay-event-normalization/src/normalize/span/tag_extraction.rs#L758) and the [description scrubbing function](https://github.com/getsentry/relay/blob/master/relay-event-normalization/src/normalize/span/description/mod.rs#L44).

### CODE REVIEW

File: relay-event-normalization/src/eap/mod.rs [hunk 1]
```diff
@@ -404,6 +408,69 @@ pub fn normalize_db_attributes(annotated_attributes: &mut Annotated<Attributes>)
...
+/// The normalization process first scrubs the url and extracts the server address from the url.
+/// It also sets 'url.full' to the raw url if it is not already set and can be retrieved from the server address.
+fn normalize_http_attributes(annotated_attributes: &mut Annotated<Attributes>) {
+    let Some(attributes) = annotated_attributes.value() else {
+        return;
+    };
+
+    // Skip normalization if not an http span.
+    // This is equivalent to conditionally scrubbing by span category in the V1 pipeline.
+    if !attributes.contains_key(HTTP_REQUEST_METHOD) {
```
Comment (Dav1dde) [inline relay-event-normalization/src/eap/mod.rs:428]: Can extend that guard to also check for presence of `SERVER_ADDRESS` or other attributes to not run it multiple times.

We do run Relay in a chain, without the guard each Relay would independently re-run the logic. If there is no clear signal whether it should run or not (e.g. maybe we want to modify existing attributes clients send), then that's fine.

File: tests/integration/test_spansv2.py [hunk 2]
```diff
@@ -1039,55 +1063,32 @@ def test_spansv2_attribute_normalization(
 
     relay.send_envelope(project_id, envelope)
 
-    assert spans_consumer.get_span() == {
-        "trace_id": "5b8efff798038103d269b633813fc60c",
-        "span_id": "eee19b7ec3c1b175",
-        "attributes": {
```
Comment (Dav1dde) [inline tests/integration/test_spansv2.py:1045]: I'd prefer if we kept asserting the full snapshot, it's a good regression test for other changes as well (e.g. maybe someone adds a new attribute to the http normalization).

To reduce the boilerplate, you can just assert the attributes and add in the common attributes vai `**common`.

File: relay-event-normalization/src/eap/mod.rs (hunk 1 above)
Comment (Ahmed-Labs) [reply relay-event-normalization/src/eap/mod.rs:428]: Oh I see! The weird thing in this case is that we read from the attribute and normalize it back into the same attribute, so we can't actually check that it was already normalized unless we write to a new field (I don't think we always want to write to a new attribute).

File: tests/integration/test_spansv2.py (hunk 2 above)
Comment (Ahmed-Labs) [reply tests/integration/test_spansv2.py:1045]: Good point, and I like the approach to reduce boilerplate, that's mainly why I didn't want to do it for each one.

**Review by Dav1dde — APPROVED** [review]

File: relay-event-normalization/src/eap/mod.rs [hunk 3]
```diff
@@ -1291,4 +1297,50 @@ mod tests {
...
+            "type": "string",
+            "value": "http.client"
+          },
+          "http.request.method": {
+            "type": "string",
+            "value": "GET"
+          },
+          "url.full": {
+            "type": "string",
+            "value": "http://192.168.1.1:3000"
```
Comment (mjq) [inline relay-event-normalization/src/eap/mod.rs]: Just a suggestion, but using a URL value that you've confirmed in another nearby test _is_ normalized when allowed hosts is empty gives a bit more confidence that normalization is being skipped for the right reason here. In other words, since `https://application.www.xn--85x722f.xn--55qx5d.cn` is used for `url.full` in `test_normalize_http_attributes`, use the same value here. Not a blocker.

Comment (Ahmed-Labs) [reply relay-event-normalization/src/eap/mod.rs]: Good point! Updated
